kore/src/main/scala/org/kframework/attributes/Att.scala

```diff
@@ -149,17 +167,25 @@ object Att {
     case object Unrecognized extends KeyType;
   }
 
+  sealed trait KeyParameter
+  private object KeyParameter extends Serializable {
+    case object Required extends KeyParameter;
+    case object Optional extends KeyParameter;
+    case object Forbidden extends KeyParameter;
+  }
+
   /* The Key class can only be constructed within Att. To enforce this, we must
    * - Make the constructor private
    * - Manually declare apply() and make it private, lest a public one is generated
    * - Manually declare copy() and make it private, preventing constructions like Att.GOOD_KEY.copy(key="bad-att")
    */
-  case class Key private[Att](key: String, keyType: KeyType) extends Serializable {
+  case class Key private[Att](key: String, keyType: KeyType, keyParam: KeyParameter) extends Serializable {
     override def toString: String = key
     private[Key] def copy(): Unit = ()
   }
   object Key {
-    private[Att] def apply(key: String, keyType: KeyType): Key = new Key(key, keyType)
+    private[Att] def apply(key: String, keyType: KeyType): Key = new Key(key, keyType, KeyParameter.Optional)
```

Comment:
It looks like you're manually passing optional each time, and this constructor is private, so let's just delete it.

Reply:
Never mind I see where this is being used now.